app/models/publisher.rb

```diff
@@ -19,6 +19,27 @@ def publish
     end
   end
 
+  def discard_draft
+    begin
+      ActiveRecord::Base.transaction do
+        publishing_api.discard_draft(content_model.content_id)
+
+        latest_published_edition = content_model.editions.published.last
+        if latest_published_edition.present?
+          content_model
+            .editions
+            .where("created_at > ?", latest_published_edition.created_at)
+            .destroy_all
+        else
+          content_model.destroy!
```

Comment:
The chances of this happening are fairly low. It would only matter if this was a guide that had never been published.

I think we should leave this for now, refactor Topic to use an actual table, and then revisit it at that point.